pages/docs/reference/python/middleware/lifecycle.mdx

```diff
@@ -0,0 +1,115 @@
+import { Callout, CodeGroup, Properties, Property, Row, Col, VersionBadge } from "src/shared/Docs/mdx";
+
+# Python middleware lifecycle
+
+The order of middleware lifecycle hooks is as follows:
+1. `transform_input`
+2. `before_memoization`
+3. `after_memoization`
+4. `before_execution`
+5. `after_execution`
+6. `transform_output`
+7. `before_response`
+
+All of these functions may be called multiple times in a single function run. For example, if your function has 2 steps then all of the hooks will run 3 times (once for each step and once for the function).
+
+Additionally, there are 2 hooks when sending events:
+1. `before_send_events`
+2. `after_send_events`
```

Comment:
I think I know the answer but just for due diligence: do you think there's anything we should add to these points that the users should know? Like, some edge cases or something that might not be intuitive?